Classes/Utility/T3jqueryUtility.php

```diff
@@ -453,7 +453,17 @@ public static function isIntegrated()
                 }
             }
         }
-        return (!$confArr['dontIntegrateOnUID'] || !is_object($GLOBALS['TSFE']) || !in_array($GLOBALS['TSFE']->id, array_values(GeneralUtility::trimExplode(',', $confArr['dontIntegrateOnUID'], true))));
+        
+        if ( ( $confArr['dontIntegrateOnUID'] && is_object($GLOBALS['TSFE']) && in_array($GLOBALS['TSFE']->id, array_values(GeneralUtility::trimExplode(',', $confArr['dontIntegrateOnUID'], TRUE))))) {
+            return FALSE;
+        }
+
+        if ( ( $confArr['dontIntegrateOnType'] && is_object($GLOBALS['TSFE']) && in_array($GLOBALS['TSFE']->type, array_values(GeneralUtility::trimExplode(',', $confArr['dontIntegrateOnType'], TRUE))))) {
```

Comment:
As check on type is typically cheaper than integration check on pageuid (possible less entries) i suggest to check for the type exclusion first